control-plane/agents/src/bin/core/node/service.rs

```diff
@@ -119,6 +119,34 @@ impl NodeOperations for Service {
         let node = self.drain(id, label).await?;
         Ok(node)
     }
+
+    /// Apply the label to node.
+    async fn label(&self, id: NodeId, label: String) -> Result<Node, ReplyError> {
+        let mut label_map = HashMap::new();
+        for pair in label.split(';') {
+            let mut iter = pair.split('=');
+            if let Some(key) = iter.next() {
+                if let Some(value) = iter.next() {
+                    label_map.insert(key.trim().to_string(), value.trim().to_string());
+                } else {
+                    return Err(SvcError::InvalidLabel { labels: label }.into());
+                }
+            } else {
+                return Err(SvcError::InvalidLabel { labels: label }.into());
+            }
+        }
+        let node = self.label(id, label_map).await?;
+        Ok(node)
+    }
+    /// Remove the specified label from the node.
+    async fn delabel(&self, id: NodeId, label: String) -> Result<Node, ReplyError> {
+        if !label.ends_with('-') || label.is_empty() {
```

Comment:
why does it need to end with `-` ? This would seem more like something we'd do at the frontend to distinguish add from remove just like in k8s plugin?